### PR TITLE
fix(`participants`): Show current user as lead participant.

### LINKED ISSUE #386: [IMPROVEMENT]: Show user at top of Participant List after participating in CrowdAction
Precondition: User participates in a CrowdAction

When user clicks on participant list, it should see itself at the top of the list to make clear he/she participates in the CrowdAction, regardless of the API's sorting method. ([api issue 107](https://github.com/CollActionteam/collaction_api/issues/107) has now been moved to app)

### PR DESCRIPTION
**Pull request by wizlif**
Closes #386

### CODE REVIEW

**Comment by wizlif** [comment]
https://user-images.githubusercontent.com/15793624/229777479-c440ca8c-ad38-4056-8b5e-3531c9e81b99.mp4

File: lib/infrastructure/participation/participation_repository.dart [hunk 1]
```diff
@@ -121,17 +120,66 @@ class ParticipationRepository implements IParticipationRepository {
...
+            participations[0] = temp;
+          }
+
+          return right(
+            PaginatedParticipations(
+              participations: participations,
+              pageInfo: pageInfo.toDomain(),
+            ),
+          );
+        }
```
Comment (Xazin) [inline lib/infrastructure/participation/participation_repository.dart:154]: What if the user isn't on the first page of participations, we should additionally check if it's on any of the latter pages and remove the user I guess.

File: lib/infrastructure/participation/participation_repository.dart [hunk 2]
```diff
@@ -121,17 +120,66 @@ class ParticipationRepository implements IParticipationRepository {
...
+      } else {
+        // Check if user exists among current participants and remove
+        int index = participations.indexWhere(
+          (p) => p.userId == authRepository.currentUser.id,
+        );
+
+        if (index > -1) {
+          participations.removeAt(index);
+        }
+      }
```
Comment (wizlif) [inline lib/infrastructure/participation/participation_repository.dart:176]: @Xazin It's handled here

Comment (Xazin) [reply lib/infrastructure/participation/participation_repository.dart:176]: It's just me who can't see which bracket belongs where, thanks Isaac haha

**Review by Xazin — APPROVED** [review]

**Pull request merged**